unittests/restart_chain_tests.cpp

```diff
@@ -205,6 +205,43 @@ BOOST_AUTO_TEST_CASE(test_existing_state_without_block_log) {
    }
 }
 
+BOOST_AUTO_TEST_CASE(restart_from_existing_state_do_not_meet_min_initial_block_num) {
+   tester chain;
+
+   std::vector<signed_block_ptr> blocks;
+   blocks.push_back(chain.produce_block());
+   blocks.push_back(chain.produce_block());
+   blocks.push_back(chain.produce_block());
+   chain.close();
+
+   auto cfg                  = chain.get_config();
+   cfg.min_initial_block_num = 100;
+   remove_existing_blocks(cfg);
+
+   // restarting chain with no block log and no genesis
```

Comment:
Can you add a test for normal cases `lib == cfg.min_initial_block_num`  and `lib > cfg.min_initial_block_num`?